tools/gepa/generate_templates.py

```
@@ -0,0 +1,398 @@
+    for ln in text.splitlines():
+        if re.match(r"^\d+\.\s+", ln.strip()):
+            pats.append(ln.strip())
+    return pats
+
+
+def get_git_baseline(root: Path, path: Path) -> Optional[str]:
+    """Try to load the baseline content from origin/main, else HEAD~1, else None."""
+    rel = path.relative_to(root).as_posix()
+    try:
+        # Prefer origin/main if available
+        out = subprocess.run(["git", "show", f"origin/main:{rel}"], capture_output=True, text=True, check=False)
```

Ruff (0.12.2)

`subprocess` call: check for execution of untrusted input

(S603)